auraed/src/runtime/cells/cell_service.rs

```diff
@@ -28,22 +28,43 @@
  *                                                                            *
 \* -------------------------------------------------------------------------- */
 
-use crate::runtime::cells::error::CellServiceError;
-use crate::runtime::cells::validation::{
+use super::validation::{
     ValidatedAllocateCellRequest, ValidatedExecutable,
     ValidatedFreeCellRequest, ValidatedStartCellRequest,
     ValidatedStopCellRequest,
 };
-use crate::runtime::cells::{Cell, CellError, CellsTable};
+use super::{Cell, CellError, CellsTable};
 use ::validation::ValidatedType;
 use aurae_proto::runtime::{
     cell_service_server, AllocateCellRequest, AllocateCellResponse,
     FreeCellRequest, FreeCellResponse, StartCellRequest, StartCellResponse,
     StopCellRequest, StopCellResponse,
 };
 use log::info;
+use thiserror::Error;
 use tonic::{Request, Response, Status};
 
+pub(crate) type Result<T> = std::result::Result<T, CellServiceError>;
+
+#[derive(Error, Debug)]
+pub(crate) enum CellServiceError {
+    #[error(transparent)]
+    CellError(#[from] CellError),
+    #[error("failed to lock cells table")]
+    FailedToObtainLock(),
+}
+
+impl From<CellServiceError> for Status {
+    fn from(err: CellServiceError) -> Self {
+        match err {
+            CellServiceError::CellError(err) => err.into(),
```

Comment:
this is still weird to me that we have one module with two error types, one of which is just translated into the other... feels like we should have one at each "service" level.

Reply:
i agree, 1 error per module.  however i add an asterisk "or service", so we should have a top-level error for Cell Service in this case.

i then think the utility classes and methods should return low-level errors with the service level translating them to the top-level error.  that's why i lean towards (1) in your list of options.

but you make a really good point about consuming Command and how that brings us a level of safety. i'm not sure how to build that into the clarity of error handling that i think the above policy would bring.